### PR TITLE
AEA-1305 fixes requested

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

* soef network check and better logging

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/connections/soef/connection.py [hunk 1]
```diff
@@ -324,9 +327,15 @@ async def _find_around_me_processor(self) -> None:
...
+            except SOEFException:
+                await self._send_error_response(
+                    oef_message,
+                    oef_search_dialogue,
+                    oef_error_operation=OefSearchMessage.OefErrorOperation.OTHER,
+                )
             except Exception:  # pylint: disable=broad-except  # pragma: nocover
                 self.logger.exception(
-                    "Exception occoured in  _find_around_me_processor"
+                    "Exception accurred in  _find_around_me_processor"
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/soef/connection.py:338]:
```suggestion
                    "Exception occurred in  _find_around_me_processor"
```

File: packages/fetchai/connections/soef/connection.py [hunk 2]
```diff
@@ -971,9 +982,21 @@ async def _stop_periodic_ping_task(self) -> None:
...
+                "get", self.base_url, timeout=self.connection_check_timeout
+            ),
+            timeout=self.connection_check_timeout,
+        )
+
     async def connect(self) -> None:
         """Connect channel set queues and executor pool."""
         self._loop = asyncio.get_event_loop()
+
+        await self._check_server_reachable()
```
Comment (DavidMinarsch) [inline packages/fetchai/connections/soef/connection.py:998]: what happens if it fails? Exception raised?

Comment (solarw) [reply packages/fetchai/connections/soef/connection.py:998]: yes

**Comment by DavidMinarsch** [comment]
Rejecting the p2p_libp2p changes as they're incompatible across platforms and Python versions. Closing and adding other changes in #2276

**Pull request closed without merging**